packages/core/src/node/messaging/messaging-contribution.ts

```diff
@@ -81,23 +83,24 @@ export class MessagingContribution implements BackendApplicationContribution, Me
 
     protected checkAliveTimeout = 30000;
     onStart(server: http.Server | https.Server): void {
-        const wss = new ws.Server({
-            server,
+        this.webSocketServer = new ws.Server({
+            noServer: true,
             perMessageDeflate: {
                 // don't compress if a message is less than 256kb
                 threshold: 256 * 1024
             }
         });
+        server.on('upgrade', this.handleHttpUpgrade.bind(this));
```

Comment:
Upgrade done on HTTP-server level instead of ws @akosyakov

Reply:
it does not change anything, I meant whether we can handle it on express level, regardless of `ws`. Your previous approach with `handleConnection` did the same in less invasive way. I would prefer it.